Gemfile

```diff
@@ -84,12 +73,31 @@ if RUBY_PLATFORM != 'java'
   end
 end
 
-# For type checking
-# Sorbet releases almost daily, with new checks introduced that can make a
-# previously-passing codebase start failing. Thus, we need to lock to a specific
-# version and bump it from time to time.
-# Also, there's no support for windows
-if RUBY_VERSION >= '2.4.0' && (RUBY_PLATFORM =~ /^x86_64-(darwin|linux)/)
-  gem 'sorbet', '= 0.5.9672'
-  gem 'spoom', '~> 1.1'
+group :appraisal do
+  gem 'appraisal', '~> 2.2'
+end
```

Comment:
### Approach 1

Very specific groups, pretty much matching the tool name: `appraisal`, `rubocop`, `yard`, `sorbet`, etc.

Reply:
This would be extremely granural, not too keen on that solution.